src/swap/utils/formatDecimals.ts

```diff
@@ -6,8 +8,54 @@ export function formatDecimals(
   inputInDecimals = true,
   decimals = 18,
 ): string {
```

Comment:
Is `throw` the correct way to approach this? And the type `string` can be missleading when returing an error.

I am leaning towards stick with our Error standard, we use in other utils and componets. So we stay predictable.

I believe we decided to `throw` only when there are must cases at config the developer has to adjust before using the library.

Reply:
this should be abstracted from the developer. `formatDecimals` is called by `getAPIParamsForToken` which is called by `buildSwapTransaction`. I wrapped the part in `buildSwapTransaction` in a try/catch and throw a `SwapError` if `formatDecimals` fails.

we will need to do a follow-up PR to handle invalid inputs in the FE component.

Reply:
An utility like `A` should always be the most pure as possible, with characteristics that makes it useful on his own. Expecting `A` doing the right thing when `B` calls `A` it creates mental complexity when in the long-run mantaining any complex library.

In a few words, we want to avoid what commonly is called "spaghetti code" which the definision says
```
 Spaghetti code is the general term used for any source code that's hard to understand because it has no defined structure. While an end user might not see anything wrong with a program, a programmer might find it virtually illegible if the code base's flow is too convoluted—like a bowl of twisted, tangled spaghetti.
```

That's why it's important to avoid at the origin this kind of slippery slope in our code.

Reply:
returned `SwapError` and moved the input validation a level up